Assets/MixedRealityToolkit/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -42,10 +46,96 @@ private static string GetPackageManifestFilePath()
+                if (versionComponents.Length == 2)
+                {
+                    // Parse the float component
+                    string prereleaseString = versionComponents[1].Trim(trimChars);
+                    if (float.TryParse(prereleaseString, out prerelease))
+                    {
+                        return true;
+                    }
+                }
+                else
+                {
+                    prerelease = 0f;
+                    return true;
+                }
+            }
+
+            version = null;
+            prerelease = float.NaN;
+            return false;
+        }
+
         /// <summary>
         /// Reports whether or not the appropriate version of MSBuild for Unity is specified
         /// in the Unity Package Manager manifest.
         /// </summary>
+        /// <param name="minPackageVersion">The minimum version of the package, as listed in the manifest.</param>
+        /// <param name="packageVersion">The version of the package, as listed in the manifest.</param>
+        /// <returns>
+        /// True if an appropriate verson of MS Build for Unity is configured in the manifest, otherwise false.
+        /// </returns>
+        internal static bool IsAppropriateMBuildVersion(string minPackageVersion, string packageVersion)
+        {
+            // Get the version of the package.
+            // Note: The version is in the following format Major.Minor.Revision[-Date.Build]
+
+            Version minVersion;
+            float minPrerelease;
+
+            // Get the min version
+            if (!TryGetVersionComponents(minPackageVersion, out minVersion, out minPrerelease))
+            {
+                return false;
+            }
+
+            // Get the current version from the manifest
+            Version currentVersion;
+            float currentPrerelease;
+            if (!TryGetVersionComponents(packageVersion, out currentVersion, out currentPrerelease))
+            {
+                return false;
+            }
+
+            // Evaluate the results.
+            // * (currentVersion > minVersion)                                                                  return true;
+            // * (currentVersion == minVersion && currentPrerelease == minPrerelease)                           return true;
+            // * (currentVersion == minVersion && minPrerelease != 0 && currentPrerelease >= minPrerelease)   return true;
+            // * all other combinatons                                                                          return false;
+            if (currentVersion > minVersion)
+            { return true; }
+            else if (currentVersion == minVersion)
+            {
+                // The current and minumum versions are the same, check the prerelease portion
+                if (currentPrerelease == minPrerelease) 
+                { return true; }
+                else if ((minPrerelease != 0f) && (currentPrerelease >= minPrerelease)) 
```

Comment:
I always get a little weirded out when I see 0 equality checks using floats because of floating point comparison error, but in C# is it safe if you're comparing it against exactly the zero literal?

Reply:
this is a parsed/preset 0 literal, so i presume so. i can do some more investigation

Reply:
based on https://docs.microsoft.com/en-us/dotnet/api/system.single.equals?view=netframework-4.8, since we are explicitly setting the value (ex: via parse) and are not calculating it, a direct check against 0 should be safe.